File: server/marketplace/src/main/java/com/dspajic/marketplace/dto/searchbar/SearchBarDto.java
package com.dspajic.marketplace.dto.searchbar;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SearchBarDto {
    private List<OptionDto> categories;
    private List<SubcategoryOptionDto> subcategories;
    private List<ProductTypeOptionDto> productTypes;
    private List<ProductOptionDto> products;
}

File: client/client/src/types/SearchBarDto.ts
export interface OptionDto {
  id: number;
  name: string;
}

export interface SubcategoryOptionDto {
    id : number;
    name : string;
    categoryId : number;
}

export interface ProductTypeOptionDto {
    id : number;
    name : string;
    categoryId : number;
    subcategoryId : number;
}

export interface ProductOptionDto {
    id : number;
    name : string;
    categoryId : number;
    subcategoryId : number;
    productTypeId: number;
}

export interface SearchBarDto {
  categories : OptionDto[];
  subcategories : SubcategoryOptionDto[];
  productTypes : ProductTypeOptionDto[];
  products : ProductOptionDto[];
}